OpenRobertaServer/staticResources/js/app/roberta/controller/galleryList.controller.js

```diff
@@ -18,14 +18,47 @@ define([ 'require', 'exports', 'log', 'util', 'comm', 'message', 'guiState.contr
      */
     function init() {
 
+        initGalleryToolbar();
         initGalleryList();
         initGalleryListEvents();
         LOG.info('init gallery list view');
     }
     exports.init = init;
 
+    function getRobotGroups(){
+        const robots = GUISTATE_C.getRobots();
+        const groups = {};
+
+        const coerceName = (name, group) => {
+            if(group === "arduino")
+                return "Nepo4Arduino";
+            if (group === "ev3")
+                return "Ev3";
+            return GUISTATE_C.getMenuRobotRealName(name)
+        }
+
+        Object.keys(robots).forEach(index => {
+            const group = robots[index].group;
+            const name = robots[index].name;
+            if(group && !groups[group]){
+                groups[group] = coerceName(name, group);
+            }
+        });
+        return groups;
+    }
+
+    function initGalleryToolbar(){
```

Comment:
Maybe you can add here the default robot as a filter?
in the current version you have still "All Robots" as default.